# HW 4 Assignment 3

Nurvirta Monarizqa (nm2773)

In [1]:
import numpy as np

# 1. Ever employed in a CEO transitional job

## NULL HYPOTHESIS: 
the % of former prisoners employed after release is the same or lower for candidates who participated in the program as for the control group, significance level p=0.05

$H_0 : P_0 - P_1 \geq 0$

$H_1 : P_0 - P_1 < 0$

$\alpha = 0.05$

In [30]:
alpha = 0.05

P_0 = 3.5 * 0.01
P_1 = 70.1 * 0.01

if P_0-P_1 >= 0:
    # we are done
    print ("the Null holds")
else:
    print ("we must assess the statistical significance")

n_0 = 409
n_1 = 564

Nt_0 = P_0 * n_0
Nt_1 = P_1 * n_1

we must assess the statistical significance


In [31]:
#define the sample proportion first
sp = (P_0 * n_0 + P_1 * n_1) / (n_1 + n_0)
print (sp)

0.4210472764645426


In [32]:
# calculate standard deviation
sp_stdev = lambda p, n: np.sqrt( p * ( 1 - p ) / n[0] +  p * ( 1 - p ) / n[1]  )


sp_stdev_2y = sp_stdev(( Nt_0 + Nt_1) / (n_0 + n_1), [n_0, n_1])
print (P_0, n_0, n_1, sp_stdev_2y)

0.035 409 564 0.0320658086057


In [33]:
# calculate z-score
zscore = lambda p0, p1, s : (p0 - p1) / s
z_2y = zscore(P_1, P_0, sp_stdev_2y)
print (z_2y)

20.7697865408


In [34]:
# 20 is way more than 3, the max of z score
# so we use 0.9998 for the p 

p_2y = 1 - 0.9998

def report_result(p,a):
    print ('is the p value {0:.2f} smaller than the critical value {1:.2f}? '.format(p,a))
    if p < a:
        print ("YES!")
    else: 
        print ("NO!")
    
    print ('the Null hypothesis is {}'.format( 'rejected' if p < a  else 'not rejected') )

    
report_result(p_2y,alpha)

is the p value 0.00 smaller than the critical value 0.05? 
YES!
the Null hypothesis is rejected


# 2. Convicted of a felony

## NULL HYPOTHESIS
the % of former prisoners recidivismed (ever convicted of a felony) after release is the same or higher for candidates who participated in the program than for the control group, significance level p=0.05

$H_0 : P1 \leq P0$

$H_1 : P1 >P0$

In [55]:
alpha = 0.05

P_0 = 11.7 * 0.01
P_1 = 10.0 * 0.01

if P_0-P_1 >= 0:
    # we are done
    print ("the Null holds")
else:
    print ("we must assess the statistical significance")

n_0 = 409
n_1 = 568

Nt_0 = P_0 * n_0
Nt_1 = P_1 * n_1

the Null holds


In [56]:
#define the sample proportion first
sp = (P_0 * n_0 + P_1 * n_1) / (n_1 + n_0)
print (sp)

0.10711668372569089


In [57]:
# calculate standard deviation
sp_stdev_rev = sp_stdev(( Nt_0 + Nt_1) / (n_0 + n_1), [n_0, n_1])
print (P_0, n_0, n_1, sp_stdev_rev)

0.11699999999999999 409 568 0.0200556791612


In [58]:
# calculate z-score
z_rev = zscore(P_1, P_0, sp_stdev_rev)
print (z_rev)

-0.84764020522


In [59]:
# for z = 0.84, P = 0.7995
p_rev = 1 - 0.7995
def report_result(p,a):
    print ('is the p value {0:.2f} smaller than the critical value {1:.2f}? '.format(p,a))
    if p < a:
        print ("YES!")
    else: 
        print ("NO!")
    
    print ('the Null hypothesis is {}'.format( 'rejected' if p < a  else 'not rejected') )

    
report_result(p_rev,alpha)

is the p value 0.20 smaller than the critical value 0.05? 
NO!
the Null hypothesis is not rejected


## Contingency table

| convicted a fellony | yes | no |
|:--:|:---:|:--:|
| test sample | 0.117 * 568 | 0.883 * 568 | 568 |
| control sample | 0.10 * 409 | 0.90 * 409 | 409 |
||||
| total | 107.356 | 869.644 |

In [61]:
chisqstat= lambda N, values, expect_num : ((values[0][0] * values[1][1]
                                            -values[0][1] * values[1][0])**2) * N / expect_num
Ntot = n_0 + n_1 # a + b + c + d = tot
tot_felony = P_0 * n_0 + P_1 * n_1
tot_non_felony = (1-P_0) * n_0 + (1-P_1) * n_1

expected_num = 568 * 409 * tot_felony * tot_non_felony
sample_values = [[P_1 * n_1, (1-P_1) * n_1], [P_0 * n_0, (1-P_0) * n_0]]
 

print (chisqstat(Ntot,  sample_values, expected_num))

0.7184939175052875


### Dof is 1 since we have 2 variables and 1 independent variable. To reject null hypothesis with alpha = 0.05, the chi-square value has to be more than 3.84 (value from the table when dof  = 1 and alpha = 0.05), but turns out the chi-square value about the conviction of fellonies above is 0.718 < 3.84. Thus, we can not reject null hypothesis.